web/html/src/manager/minion/packages/package-states.js

```diff
@@ -1,399 +1,363 @@
-          console.log(data);
-          this.setState({
-            view: "search",
-            search:  {
-                filter: this.state.filter,
-                results: data.map(state => {
-                  state.packageStateId = normalizePackageState(state.packageStateId);
-                  return state;
-                })
-            }
-          });
-        });
+      setChanged(draft => {
+        draft[key] = {
+          original: original,
+          value: {
+            arch: original.arch,
+            epoch: original.epoch,
+            version: original.version,
+            release: original.release,
+            name: original.name,
+            packageStateId: newPackageStateId,
+            versionConstraintId: newVersionConstraintId
+          }
+        };
+      });
     }
   }
 
-  save() {
-    const states = [];
-    for(var state of this.state.changed.values()) {
-        states.push(state.value)
+  const applyPackageState = () => {
+    if (changed.size > 0) {
+      const response = confirm(t("There are unsaved changes. Do you want to proceed ?"));
+      if (response === false) {
+        return null;
+      }
     }
-    const request = Network.post(
-        "/rhn/manager/api/states/packages/save",
-        JSON.stringify({
-            sid: serverId,
-            packageStates: states
-        }),
-        "application/json"
-    ).promise.then(data => {
-      console.log("success: " + data);
-      const newPackageStates = data.map(state => {
-          state.packageStateId = normalizePackageState(state.packageStateId);
-          return state;
-      });
 
-      const newSearchResults = this.state.search.results.map( state => {
-        const changed = this.state.changed.get(packageStateKey(state))
-        if(changed !== undefined) {
-            return changed.value;
-        } else{
-            return state;
-        }
-      });
-
-      this.setState({
-        changed: new Map(),
-        view: "system",
-        search: {
-            filter: this.state.search.filter,
-            results: newSearchResults
-        },
-        packageStates: newPackageStates,
-        messages: MessagesUtils.info(t('Package states have been saved.'))
-      });
-    }, jqXHR => {
-      console.log("fail: " + jqXHR);
-      throw "failed to save";
+    fetchPackageStatesApi(action.APPLY, serverId)
+      .then(data => {
+        console.log("apply action queued:" + data);
+      }).catch(error => {
+      showErrorToastr(error, {autoHide: false});
     });
-    return request;
-  }
+  };
 
-  applyPackageState() {
-    if (this.state.changed.size > 0) {
-        const response = confirm(t("There are unsaved changes. Do you want to proceed ?"))
-        if (response == false) {
-            return null;
-        }
+  const save = (): Promise<any> => {
+    const toSave = [];
```

Comment:
I think we could simplify this as:

```javascript
const changedValues = Object.values(changed)
  .filter(changedValue => changedValue.value)
  .map(changedValue => changedValue.value)
```

the changedValue.value should always be an object if defined, that's why we have a type checker as Flow.  Am I overlooking something here?